kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java

```diff
@@ -205,18 +203,28 @@ public SubscriptionPackMessage nextScheduledMessage(DateTime date) { //NO CHECKS
                     String.format("Subscription with ID %s is not due for any scheduled message. Start date in the future", subscriptionId));
         }
 
-        if (!DayOfTheWeek.fromDateTime(date).equals(firstMessageDayOfWeek) && !DayOfTheWeek.fromDateTime(date).equals(secondMessageDayOfWeek)) {
-            // bad call, no message scheduled for the day
-            throw new IllegalStateException(String.format("Subscription with ID %s is not due for any scheduled message for given date", subscriptionId));
+        // If completed, send last message in pack for 7 day window after completion
+        if (status == SubscriptionStatus.COMPLETED) {
+            if (daysIntoPack < (subscriptionPack.getWeeks() * DAYS_IN_WEEK + DAYS_IN_WEEK)) {
+                return getMessageByWeekAndMessageId(subscriptionPack.getWeeks(), subscriptionPack.getMessagesPerWeek());
+            } else {
+                throw new IllegalStateException("No inbox found for completed subscription");
```

Comment:
"Trying to get last subscription message more than a week after the subscription has completed."